test_crates/struct_now_doc_hidden/new/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+mod MyNonPublicMod {
+    // despite adding #[doc(hidden)], this struct is in a
+    // private mod, so it isn't part of the crate's public
+    // api
+    #[doc(hidden)]
+    pub struct MyStruct;
+}
+
+pub mod MyPublicMod {
+    // added #[doc(hidden)], however this struct is in a
+    // public mod, so it is part of the crate's public api
+    #[doc(hidden)]
+    pub struct MyStruct;
+}
+
+mod MyNestedNonPublicMod {
+    pub mod PublicInnerStruct {
+        // despite adding #[doc(hidden)], this struct is in a
+        // private outer mod, so it isn't part of the crate's public
+        // api
+        #[doc(hidden)]
+        pub struct MyStruct;
+    }
+}
+
+pub mod MyNestedPublicMod {
+    pub mod PublicInnerStruct {
+        // added #[doc(hidden)], however this struct is in a
+        // public mod, so it is part of the crate's public api
+        #[doc(hidden)]
+        pub struct MyStruct;
+    }
+}
+
+#[doc(hide)] // shouldn't flag, it should be #[doc(hidden)] not #[doc(hide)]
```

Comment:
I don't think this is a legal attribute, and it might break in the future. Here's the list of things the `doc` attribute supports and we probably should stick to it: https://doc.rust-lang.org/rustdoc/write-documentation/the-doc-attribute.html

I'd probably pick one of the other existing `doc` commands instead of using one that doesn't exist.

Reply:
good point, I replaced it with a few valid doc attribute values.